### PR TITLE
Fix atsign activation issue

### PR DESCRIPTION
**Pull request by sachins-geekyants**
**- What I did**
- Added fix for [#381](https://github.com/atsign-company/at_handle_registrar_web/issues/381), [#659](https://github.com/atsign-foundation/apps/issues/659)

**- How I did it**
- Validate person call was missing for one of the test scenarios.
- Added a delay when atsign is moved from unavailable state to teapot state.

### CODE REVIEW

File: at_onboarding_flutter/lib/screens/pair_atsign.dart [hunk 1]
```diff
@@ -773,7 +779,9 @@ class _PairAtsignWidgetState extends State<PairAtsignWidget> {
             builder: (_) => WillPopScope(
               onWillPop: () async {
                 int ct = 0;
-                Navigator.of(context).popUntil((_) => ct++ >= 2);
+                Navigator.of(context).popUntil((_) {
+                  return ct++ >= 2;
```
Comment (yahu1031) [inline at_onboarding_flutter/lib/screens/pair_atsign.dart:783]: Can you explain why this is? And if not necessary, please remove it.

**Review by nitesh2599 — APPROVED** [review]

File: at_onboarding_flutter/lib/widgets/custom_dialog.dart [hunk 2]
```diff
@@ -1143,6 +1147,21 @@ class _CustomDialogState extends State<CustomDialog> {
               cramSecret = await validatePerson(value, email, otp, context,
                   isConfirmation: true);
               return cramSecret;
+            } else if (value != null) {
+              AtStatusImpl atStatusImpl =
+                  AtStatusImpl(rootUrl: AppConstants.serverDomain);
+              AtStatus status = await atStatusImpl.get(value);
+
+              /// if atsign is not yet activated, it will be activated.
+              if (status.atSignStatus == AtSignStatus.unavailable &&
```
Comment (yahu1031) [inline at_onboarding_flutter/lib/widgets/custom_dialog.dart:1156]: You can remove `atSignStatus` comparison. Cause that was returning `unavailable` for any atsign. Reference [Image 1](https://cdn.discordapp.com/attachments/756903745241088011/993118732442415244/unknown.png), [Image 2](https://cdn.discordapp.com/attachments/756903745241088011/993118753845936178/unknown.png)

**Review by yahu1031 — COMMENTED** [review]
These are my findings that are NLGTM.

File: at_onboarding_flutter/lib/widgets/custom_dialog.dart (hunk 2 above)
Comment (sachins-geekyants) [reply at_onboarding_flutter/lib/widgets/custom_dialog.dart:1156]: Added this check to confirm atSign is in unactivated state.
Even I noticed, atSign always remains in unactivated state, looks like a bug in at_server package.

File: at_onboarding_flutter/lib/screens/pair_atsign.dart (hunk 1 above)
Comment (sachins-geekyants) [reply at_onboarding_flutter/lib/screens/pair_atsign.dart:783]: I don't completely understand it's behaviour, haven;t made any changes to it. Just converted it from the arrow function to block function.

File: at_onboarding_flutter/lib/widgets/custom_dialog.dart (hunk 2 above)
Comment (yahu1031) [reply at_onboarding_flutter/lib/widgets/custom_dialog.dart:1156]: Yes, I can take a look there

File: at_onboarding_flutter/lib/screens/pair_atsign.dart (hunk 1 above)
Comment (yahu1031) [reply at_onboarding_flutter/lib/screens/pair_atsign.dart:783]: Yes, we need to provide a false statement there. We can either pass false there.

File: at_onboarding_flutter/lib/widgets/custom_dialog.dart (hunk 2 above)
Comment (yahu1031) [inline at_onboarding_flutter/lib/widgets/custom_dialog.dart:1156]: use `status.atSignStatus.status()` to get correct status.

Comment (yahu1031) [reply at_onboarding_flutter/lib/widgets/custom_dialog.dart:1156]: Or else here is the fix https://github.com/atsign-foundation/at_libraries/pull/190

**Pull request merged**